codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ServerServiceGeneratorV2.kt

```diff
@@ -469,8 +472,108 @@ class ServerServiceGeneratorV2(
     }
 
     fun render(writer: RustWriter) {
+        val crateName = codegenContext.moduleUseName()
+        val handlers: Writable = operations
+            .map { operation ->
+                DocHandlerGenerator(operation, "///", builderFieldNames[operation]!!, codegenContext).docSignature()
+            }
+            .reduce { acc, wt ->
+                writable {
+                    rustTemplate("#{acc:W} \n#{wt:W}", "acc" to acc, "wt" to wt)
+                }
+            }
+
+        val hasErrors = service.operations.any { model.expectShape(it).asOperationShape().get().errors.isNotEmpty() }
+
         writer.rustTemplate(
             """
+            /// A fast and customizable Rust implementation of the $serviceName Smithy service.
+            ///
+            /// The primary export is [`$serviceName`]: it satisfies the [`Service<http::Request, Response = http::Response>`]
+            /// trait and therefore can be handed to [Hyper server] using [`$serviceName::into_make_service`] or used in Lambda using [`#{SmithyHttpServer}::routing::LambdaHandler`].
+            /// The [`crate::${InputsModule.name}`], ${if (!hasErrors) "and " else ""}[`crate::${OutputsModule.name}`], ${if (hasErrors) "and [`crate::${ErrorsModule.name}`]" else "" }
+            /// modules provide the types used in each operation.
+            ///
+            /// The primary export is [`$serviceName`]: it is the
+            /// [`$builderName`] is used to set your business logic to implement your [operations],
+            /// customize your [operations]'s behaviors by applying middleware,
+            /// and build your service.
+            ///
+            /// [`$builderName`] contains the [operations] modeled in your Smithy service.
+            /// You must set an implementation for all operations with the correct signature,
+            /// or your service will fail to be constructed at runtime and panic.
+            /// For each of the [operations] modeled in
+            /// your Smithy service, you need to provide an implementation in the
+            /// form of an async function that takes in the
+            /// operation's input as their first parameter, and returns the
+            /// operation's output. If your operation is fallible (i.e. it
+            /// contains the `errors` member in your Smithy model), the function
+            /// implementing the operation has to be fallible (i.e. return a [`Result`]).
+            /// The possible forms for your async functions are:
+            /// ```rust,no_run
+            /// async fn handler_fallible(input: Input, extensions: #{SmithyHttpServer}::Extension<T>) -> Result<Output, Error>
+            /// async fn handler_infallible(input: Input, extensions: #{SmithyHttpServer}::Extension<T>) -> Output
+            /// ```
+            /// Both can take up to 8 extensions, or none:
```

Comment:
We should probably call these "extractors" to be in line with `axum` and other HTTP backend frameworks, rather than extensions which collide with `Extension<T>` (which is an extractor ofc).